src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp

```diff
@@ -0,0 +1,542 @@
+                                        "maxEnergy[keV]",
+                                        &maxEnergy_keV);
+
+            hdf5DataFile.writeAttribute(currentStep,
+                                        SplashTypeBool,
+                                        "detector",
+                                        "logScale",
+                                        &this->logScale);
+        }
+
+        hdf5DataFile.close();
+    }
+
+public:
+    ParticleDetector() :
+        name("ParticleDetector: (virtually) propagates and collects particles to infinite distance"),
+        prefix(ParticlesType::FrameType::getName() + std::string("_detector")),
+        foldername(prefix),
+        notifyPeriod(0),
+        cellDescription(NULL),
+        dBufCalorimeter(NULL),
+        dBufLeftParsCalorimeter(NULL),
+        hBufCalorimeter(NULL),
+        hBufTotalCalorimeter(NULL)
+    {
+        Environment<>::get().PluginConnector().registerPlugin(this);
+    }
+
+
+    void notify(uint32_t currentStep)
+    {
+        /* initialize detector with already detected particles */
+        *this->dBufCalorimeter = *this->dBufLeftParsCalorimeter;
+
+        /* data is written to dBufCalorimeter */
+        this->calorimeterFunctor->setCalorimeterCursor(this->dBufCalorimeter->origin());
+
+        /* create kernel functor instance */
+        DataConnector &dc = Environment<>::get().DataConnector();
+        ParticlesType* particles = &(dc.getData<ParticlesType > (ParticlesType::FrameType::getName(), true));
+
+        ParticleDetectorKernel<typename ParticlesType::ParticlesBoxType,
+                               MyCalorimeterFunctor> particleDetectorKernel(
+            particles->getDeviceParticlesBox(),
+            *this->calorimeterFunctor);
+
+        /* create zone */
+        typedef typename MappingDesc::SuperCellSize SuperCellSize;
+
+        const PMacc::math::Int<simDim> coreBorderGuardSuperCells = this->cellDescription->getGridSuperCells();
+        const uint32_t guardSuperCells = this->cellDescription->getGuardingSuperCells();
+        const PMacc::math::Int<simDim> coreBorderSuperCells = coreBorderGuardSuperCells - 2*guardSuperCells;
+
+        /* this zone represents the core+border area with guard offset in unit of cells */
+        const zone::SphericZone<simDim> zone(
+            static_cast<PMacc::math::Size_t<simDim> >(coreBorderSuperCells * SuperCellSize::toRT()),
+            guardSuperCells * SuperCellSize::toRT());
+
+        /* kernel call */
+        algorithm::kernel::Foreach<SuperCellSize> foreach;
+        foreach(zone, cursor::make_MultiIndexCursor<simDim>(), particleDetectorKernel);
+
+        /* copy to host */
+        *this->hBufCalorimeter = *this->dBufCalorimeter;
+
+        /* mpi reduce */
+        using namespace lambda;
+        (*this->allGPU_reduce)(*this->hBufTotalCalorimeter, *this->hBufCalorimeter, _1 + _2);
+        if(!this->allGPU_reduce->root())
+            return;
```

Comment:
you are using the same lines of code [here](https://github.com/Heikman/picongpu/blob/bba18337b644a5e453e176605f175e3a26b00fec/src/picongpu/include/plugins/particleDetector/ParticleDetector.hpp#L171-L175) as well.